boards/arm/96b_argonkey/96b_argonkey.dts

```diff
@@ -60,7 +60,7 @@
 
 	/* ST Microelectronics LSM6DSL accel/gyro sensor */
 	lsm6dsl@1 {
-		compatible = "st,lsm6dsl-spi";
+		compatible = "st,lsm6dsl";
 		reg = <1>;
 		spi-max-frequency = <1000000>;
```

Comment:
IIUC, this whole change allows to have these bus related properties documented by the binding depending on the bus. Is that compatible with device tree usage in eg Linux?

I'm still not convinced how this would work at driver level. To me we should be able to build matching sensor driver using "compatible".
When "compatible" carried SPI/I2C inforamtion, we could have used #idef DT_COMPAT_ST_LSM6DL_SPI / DT_COMPAT_ST_LSM6DL_I2C to know if driver should be compiled with SPI or I2C fields.
Now this information is no more present we'll need to have a new piece of info to know how to build the driver with fields matching the bus.

Reply:
but then it should be available as a compilation input, as a configuration input.

Reply:
The bus type can be inferred by looking at the parent node, and the tooling generate the appropriate flags.  The BME280 supports SPI and I2C, and the Linux binding documentation doesn't represent which is being used: it just provides two drivers, each of which are compatible with `bosch,bme280`.

The inference would be simplified by marking all I2C bus nodes as compatible with `zephyr,i2c` and similarly for other generic Zephyr drivers.  This is something I'd like done regardless.

Reply:
> IIUC, this whole change allows to have these bus related properties documented by the binding depending on the bus. Is that compatible with device tree usage in eg Linux?

Yes, if you look at how various sensor drivers in linux you'll see they use the same compatible regardless of what bus the device is on.

Reply:
> The inference would be simplified by marking all I2C bus nodes as compatible with `zephyr,i2c` and similarly for other generic Zephyr drivers. This is something I'd like done regardless.

I don't see us ever doing this, as we should be aiming for the device tree's to have less zephyr specific details not more.  The intent would be to know that a bus is I2C or SPI via the binding YAML info.